### ISSUE TITLE
Add lazy multidimensional Unit terms

### ISSUE THREAD

**Issue by eb8680** [pyro-ppl/funsor#235]
It would be useful to have unit terms corresponding to deferred calls to `torch.zeros`/`torch.ones` that conform to constant propagation rules #163 .  This would improve cons hashing accuracy and allow us to avoid reifying large unit tensors as can sometimes happen in e.g. `funsor.adjoint`.

**Comment by eb8680** [comment]
I'm no longer convinced this is necessary - we should never need to `expand` constants.
